### PR TITLE
feat(SafeShield): Malicious adresses in Shield threat review

### PR DESCRIPTION
**Pull request by vmosharova**
## What it solves
Displays malicious addresses from threat analysis issues in the UI. When the backend returns threat analysis results with issues containing addresses (e.g., `{description: "Malicious contract interaction", address: "0x1234..."}`), these addresses were not shown in the UI.

Resolves:[ COR-937](https://linear.app/safe-global/issue/COR-937/frontend-display-description-and-address-for-critical-thread-analysis)

The changes are implemented to reflect the API schema change in the BE ([2824](https://github.com/safe-global/safe-client-gateway/pull/2824) and [2829](https://github.com/safe-global/safe-client-gateway/pull/2829) 

## How this PR fixes it

1. **Type definitions updated**:
   - Added `ThreatIssue` type with `{description: string, address?: string}` structure
   - Updated `MaliciousOrModerateThreatAnalysisResult` to use `ThreatIssue[]` instead of `string[]` for issues
   - Updated auto-generated types to match the backend DTO format

2. **Transformation logic created**:
   - Added `transformThreatAnalysisResponse` utility that extracts addresses from `ThreatIssue` objects in the `issues` property
   - Extracted addresses are added to the `addresses` array of each threat result
   - Prevents duplicate addresses and preserves existing addresses

3. **Hook integration**:
   - Updated `useThreatAnalysis` to use `transformThreatAnalysisResponse` to process threat data before returning it
   - Ensures addresses from issues are available to UI components

4. **Web + Mobile update**:
The components + UI are updated both for Web and Mobile.

5. **Test coverage**:
   - Added tests covering address extraction, duplicate prevention, UI changes, and edge cases

## Screenshots 
image

## How to test it

1. **Manual testing**:
   - Create a transaction that triggers a threat analysis with issues containing addresses
   - Verify that the threat analysis section shows:
     - Addresses are clickable and show explorer links
     
## Checklist

- [x] I've tested the branch on mobile 📱
- [x] I've documented how it affects the analytics (if at all) 📊 - No analytics changes
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻 - Added tests for `transformThreatAnalysisResponse`

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

---

> [!NOTE]
> Displays addresses from threat analysis issues across web and mobile by transforming API responses and updating UI to show/copy/open addresses with conditional address lists.
> 
> - **Utils/Types**:
>   - Add `ThreatIssue` (`{ description, address? }`) and update threat `issues` typing.
>   - Introduce `transformThreatAnalysisResponse` to extract issue addresses into `result.addresses`; integrate in `useThreatAnalysis`.
>   - Update builders to include sample addresses in malicious/moderate presets.
> - **Mobile UI (SafeShield)**:
>   - `AnalysisDisplay`: render `AnalysisIssuesDisplay` when `issues` exist; otherwise show `ShowAllAddress`.
>   - New `AnalysisPaper` wrapper and `useAnalysisAddress` hook (copy-to-clipboard, explorer open).
>   - Refactor `AddressListItem` and `ShowAllAddress` to use new hook and wrapper; minor key fixes in `AnalysisGroup`.
> - **Web UI (SafeShield)**:
>   - `AnalysisGroupCardItem`: only show `ShowAllAddress` when no `issues`; always render `AnalysisIssuesDisplay` and address changes as applicable.
>   - `AnalysisIssuesDisplay`: show issue address (copy + optional explorer link) with description; ignore empty issues.
> - **Tests**:
>   - Add comprehensive tests for transformer, web and mobile issue/address rendering, and conditional address list behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e0b44f36a4b5da28218be09943d514aab684471c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by vmosharova** [comment]
@codex review this draft

**Comment by vmosharova** [comment]
@codex review

File: packages/utils/src/features/safe-shield/utils/transformThreatAnalysisResponse.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    return baseResult
+  })
+
+  return {
+    ...(transformedThreats && transformedThreats.length > 0 ? { THREAT: transformedThreats } : {}),
+    ...(response.BALANCE_CHANGE && response.BALANCE_CHANGE.length > 0
+      ? { BALANCE_CHANGE: response.BALANCE_CHANGE }
+      : {}),
```
[the remark this answers is not in the document]
Comment (vmosharova) [reply packages/utils/src/features/safe-shield/utils/transformThreatAnalysisResponse.ts:67]: Fixed

File: apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx [hunk 2]
```diff
@@ -21,14 +33,30 @@ export function AnalysisIssuesDisplay({ result }: AnalysisIssuesDisplayProps) {
...
+        issues.map((issue, index) => {
+          const explorerLink =
+            activeChain?.blockExplorerUriTemplate &&
+            getExplorerLink(issue.address ?? '', activeChain.blockExplorerUriTemplate)
+
+          return (
+            <AnalysisPaper key={`${severity}-${index}`} spaced={Boolean(explorerLink)}>
+              {issue.address && (
+                <AddressListItem
+                  index={index}
```
[the remark this answers is not in the document]
Comment (vmosharova) [reply apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx]: Fixed ( [5ab7555](https://github.com/safe-global/safe-wallet-monorepo/pull/6776/commits/5ab7555f6e0e5fb0f3e1a4ea9a34008ca27ab04b) )

**Review by katspaugh — APPROVED** [review]
LGTM! It would be good to break down mobile/web parts into separate PRs next time.

File: apps/web/src/features/safe-shield/components/AnalysisIssuesDisplay/AnalysisIssuesDisplay.tsx [hunk 3]
```diff
@@ -4,30 +4,108 @@ import type {
...
+                  </Box>
+                </Typography>
+              )}
 
-        <Typography variant="body2" color="primary.light" fontStyle="italic">
-          {issue.description}
-        </Typography>
-      </Box>
-    )),
+              {issue.description && (
```
Comment (LucieFaire) [inline apps/web/src/features/safe-shield/components/AnalysisIssuesDisplay/AnalysisIssuesDisplay.tsx]: if array of issues isn't empty - the description should always be present, so maybe this check is obsolete

Comment (vmosharova) [reply apps/web/src/features/safe-shield/components/AnalysisIssuesDisplay/AnalysisIssuesDisplay.tsx]: Fixed :)

**Comment by LucieFaire** [comment]
love the new design! 💟

File: apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/AnalysisDisplay.tsx [hunk 4]
```diff
@@ -55,11 +55,13 @@ export function AnalysisDisplay({ result, description, severity }: AnalysisDispl
...
-          <AnalysisIssuesDisplay result={result} />
-
           {isAddressChange(result) && <AddressChanges result={result} />}
 
-          {result.addresses?.length && <ShowAllAddress addresses={result.addresses.map((a) => a.address)} />}
+          {'issues' in result ? (
+            <AnalysisIssuesDisplay result={result} />
+          ) : result.addresses?.length ? (
+            <ShowAllAddress addresses={result.addresses.map((a) => a.address)} />
+          ) : null}
```
[the remark this answers is not in the document]
Comment (vmosharova) [reply apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/AnalysisDisplay.tsx:72]: Done

File: apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx [hunk 5]
```diff
@@ -4,31 +4,62 @@ import type {
...
-            <Text fontSize="$4" color="$colorLight" fontStyle="italic" flex={1}>
-              {issue.description}
-            </Text>
-          </View>
-        )),
+        issues.map((issue, index) => {
+          const globalIndex = issueCounter++
+          const explorerLink =
+            activeChain?.blockExplorerUriTemplate &&
+            getExplorerLink(issue.address ?? '', activeChain.blockExplorerUriTemplate)
```
[the remark this answers is not in the document]
Comment (vmosharova) [reply apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx]: Done

File: apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx [hunk 6]
```diff
@@ -4,31 +4,62 @@ import type {
...
+  const { handleOpenExplorer, handleCopyToClipboard, copiedIndex } = useAnalysisAddress()
+
   if (!('issues' in result)) {
     return null
   }
 
   const issues = result.issues as MaliciousOrModerateThreatAnalysisResult['issues']
   const sortedIssues = sortByIssueSeverity(issues)
 
+  let issueCounter = 0
```
[the remark this answers is not in the document]
Comment (vmosharova) [reply apps/mobile/src/features/SafeShield/components/AnalysisGroup/AnalysisDisplay/components/AnalysisIssuesDisplay.tsx:39]: Done

**Review by compojoom — APPROVED** [review]
